packages/react-storage/src/components/StorageBrowser/views/Controls/Navigate.tsx

```diff
@@ -115,8 +115,9 @@ export function NavigateControl({
       </NavigateItem>
       {previous?.map((destination, index) => {
         const { bucket, id, prefix: _prefix } = destination;
-        // remove trailing `/` from `prefix`
-        const prefix = _prefix?.endsWith('/') ? _prefix.slice(0, -1) : _prefix;
+
+        // remove trailing `/` from `prefix` and use only the last segment since we only want current location
+        const prefix = _prefix?.replace(/\/$/, '').split('/').pop() ?? '';
```

Comment:
This change is required to show the correct location when navigating to nested folders. It extracts only the last segment and removes the trailing `/`, if nothing is present falls back to empty string.

Reply:
How is this related to the folder uploads bug?

Reply:
+1. Don't think we need this and also think it might be getting touched in https://github.com/aws-amplify/amplify-ui/pull/5990

Reply:
Got it,  I thought it was due to my change, if it exists regardless I won't update it.